kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardHolderLoadServiceImpl.java

```diff
@@ -0,0 +1,179 @@
+            LOG.error("error while getting file bytes:  " + e.getMessage(), e);
+            throw new RuntimeException("Error encountered while attempting to get file bytes: " + e.getMessage(), e);
+        } catch (ParseException e) {
+            LOG.error("Error parsing xml " + e.getMessage());
+            throw new RuntimeException("Error parsing xml " + e.getMessage(), e);
+        }
+
+        if (pcardHolders == null || pcardHolders.isEmpty()) {
+            LOG.warn("No PCard Holders in input file " + fileName);
+        }
+
+        loadTransactions((List<ProcurementCardHolderLoad>) pcardHolders);
+
+        LOG.info("Total holders loaded: " + pcardHolders.size());
+        return true;
+    }
+    
+    @Override
+    public boolean loadFiles() {
+        cleanTransactionsTable();
+
+        List<String> fileNamesToLoad = batchInputFileService.listInputFileNamesWithDoneFile(procurementCardHolderInputFileType);
+
+        //process only most recent file set to prevent duplicates
+        Map<Date, String> fileNameMap = buildMapForFileNameList(fileNamesToLoad);
+        Date holdDate = null;
+        for (Iterator<Date> iter2 = fileNameMap.keySet().iterator(); iter2.hasNext();) {
+            Date fileDate = (Date) iter2.next();
+            if (holdDate == null || holdDate.compareTo(fileDate) < 0) {
+                holdDate = fileDate;
+            }
+        }
+        String holdDateString = (String) fileNameMap.get(holdDate);
+        
+        boolean processSuccess = true;
+
+        for (String inputFileName : fileNamesToLoad) {
+            if (inputFileName.contains(holdDateString)) {
+                // Removing related .done file right away
+                removeDoneFile(inputFileName);
+                LOG.info("Loading Procurement Cardholder file " + new File(inputFileName).getName());
+                processSuccess = loadProcurementCardHolderFile(inputFileName);
+            }
+        }
+        //remove any leftover done files
+        removeDoneFiles(fileNamesToLoad);
+        return processSuccess;
+    }
+    
+    private Map<Date, String> buildMapForFileNameList(List<String> fileNamesToLoad) {
+        Map<Date, String> fileNameMap = new HashMap<Date, String>();
+        File inputFile = null;
+        String shortFileName = null;
+        
+        for (String inputFileName : fileNamesToLoad) {
+            inputFile = new File(inputFileName);
+            shortFileName = inputFile.getName();
+            // place file date values in map
+            fileNameMap.put(Date.valueOf(shortFileName.substring(10, 20)), shortFileName.substring(10, 20));
+        }
+        return fileNameMap;
+    }
+    
+    /**
+     * Clears out associated .done files for the processed data files.
+     */
+    private void removeDoneFiles(List<String> dataFileNames) {
+        for (String dataFileName : dataFileNames) {
+            File doneFile = new File(StringUtils.substringBeforeLast(dataFileName, ".") + ".done");
```

Comment:
Replace this code with invoking the other method to avoid code duplication:
removeDoneFile(dataFileName)